acceptance/tests/allow_arbitrary_node_name_fact_for_agent.rb

```diff
@@ -11,9 +11,109 @@
 
 node_names.uniq!
 
-authfile = "#{testdir}/auth.conf"
-authconf = node_names.map do |node_name|
-  %Q[
+if @options[:is_puppetserver]
+  step "Ensure legacy auth.conf is disabled for test" do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => false}})
+  end
+
+  teardown do
+    modify_tk_config(master, options['puppetserver-config'], {'jruby-puppet' => {'use-legacy-auth-conf' => true}})
+  end
+
+  step "Backup the tk auth.conf file" do
+    on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.conf /etc/puppetlabs/puppetserver/conf.d/auth.bak'
+  end
+
+  teardown do
+    # restore the original tk auth.conf file
+    on master, 'cp /etc/puppetlabs/puppetserver/conf.d/auth.bak /etc/puppetlabs/puppetserver/conf.d/auth.conf'
+  end
+
+  step "Setup tk-auth rules" do
+    tka_header = <<-HEADER
+authorization: {
+    version: 1
+    rules: [
+        {
+            match-request: {
+                path: "/puppet/v3/file"
+                type: path
+            }
+            allow: "*"
+            sort-order: 500
+            name: "puppetlabs file"
+        },
+    HEADER
+
+    tka_node_rules = node_names.map do |node_name|
+      <<-NODE_RULES
+        {
+            # Allow nodes to retrieve their own catalog
```

Comment:
These comment lines seem a bit misleading.  I think these are allowing all users to retrieve the `.../node_name` endpoint - as opposed to just the `node_name` user to retrieve its own endpoint.  Maybe could just take these comment lines out since the rules otherwise seem self-describing.